src/brain.ino

```diff
@@ -79,36 +120,132 @@ void setup() {
   pinMode(txPin, OUTPUT);
   digitalWrite(txPin, LOW);
 
-  // delay(4000);
-  //
-  // char cd_changer_hello[] = "1100001101001000000100";
-  // String head_response1 = "111011110100110010";
-  // String head_response2 = "111001110100110011";
-  // String ideal_head_response = "111001101110000100";
-  // char yeah[] = "110000110100000011";
-  // boolean unmatched = true;
-  //
-  // send_message(cd_changer_hello);
-  //
-  // while(unmatched) {
-  //   response = listen_for_message();
-  //   Serial.println(response);
-  //
-  //   if(response == head_response1) {
-  //     unmatched = false;
-  //     send_message(cd_changer_hello);
-  //   } else if(response == head_response2) {
-  //     unmatched = false;
-  //     send_message(cd_changer_hello);
-  //   } else if(response == ideal_head_response) {
-  //     unmatched = false;
-  //     send_message(yeah);
-  //   }
-  // }
+  boolean unmatched = true;
+
+  delay(4000);
+
+  // listen for head1 until we hear it
+  char head1[] = "1101010100";
+  while(unmatched) {
+    response = listen_for_message();
+
+    if(response == head1) {
+      unmatched = false;
+    }
+  }
```

Comment:
I think this is a good first start. If we don't get the message, or if we only get head1, but not head2, we'd see it hang here :+1: